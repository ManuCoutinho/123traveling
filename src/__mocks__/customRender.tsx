import { render } from '@testing-library/react'
import { RenderOptions } from '@testing-library/react/types'
import { ChakraProvider } from '@chakra-ui/react'
import { ReactElement, FC, ReactNode } from 'react'

const MockProviders: FC<{ children: ReactNode }> = ({ children }) => {
	return <ChakraProvider>{children}</ChakraProvider>
}

const customRender = (
	ui: ReactElement,
	options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: MockProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
