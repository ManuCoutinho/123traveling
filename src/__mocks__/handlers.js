import { rest } from 'msw'

export const handlers = [
	rest.get(
		'https://api.unsplash.com/photos/nxkinK7hhOk',
		async (req, res, ctx) => {
			return res(
				ctx.status(200),
				ctx.json({
					full: 'https://images.unsplash.com/photo-1629984557780-4dde2292e245?crop=entropy\u0026cs=srgb\u0026fm=jpg\u0026ixid=MnwzMTA0MzV8MHwxfGFsbHx8fHx8fHx8fDE2ODI1OTYyODU\u0026ixlib=rb-4.0.3\u0026q=85',
					regular:
						'https://images.unsplash.com/photo-1629984557780-4dde2292e245?crop=entropy\u0026cs=tinysrgb\u0026fit=max\u0026fm=jpg\u0026ixid=MnwzMTA0MzV8MHwxfGFsbHx8fHx8fHx8fDE2ODI1OTYyODU\u0026ixlib=rb-4.0.3\u0026q=80\u0026w=1080',
					thumb:
						'https://images.unsplash.com/photo-1629984557780-4dde2292e245?crop=entropy\u0026cs=tinysrgb\u0026fit=max\u0026fm=jpg\u0026ixid=MnwzMTA0MzV8MHwxfGFsbHx8fHx8fHx8fDE2ODI1OTYyODU\u0026ixlib=rb-4.0.3\u0026q=80\u0026w=200',
					small:
						'https://images.unsplash.com/photo-1629984557780-4dde2292e245?crop=entropy\u0026cs=tinysrgb\u0026fit=max\u0026fm=jpg\u0026ixid=MnwzMTA0MzV8MHwxfGFsbHx8fHx8fHx8fDE2ODI1OTYyODU\u0026ixlib=rb-4.0.3\u0026q=80\u0026w=400',
					author: 'Bruno Thethe',
					alt: 'gray concrete bridge over river',
					userLink: 'https://unsplash.com/@bruthethe',
					downloadLocation:
						'https://api.unsplash.com/photos/nxkinK7hhOk/download?ixid=MnwzMTA0MzV8MHwxfGFsbHx8fHx8fHx8fDE2ODI1OTYyODU'
				})
			)
		}
	)
]
