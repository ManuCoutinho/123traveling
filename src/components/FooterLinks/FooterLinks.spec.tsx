import { screen } from '@testing-library/react'
import { render } from '__mocks__/customRender'
import { FooterLinks } from '.'

const mockProps = {
	links: [
		{
			listItem: 'item 1',
			url: '#',
			id: '001',
			text: 'description1'
		},
		{
			listItem: 'item 2',
			url: '#',
			id: '002',
			text: 'description2'
		}
	],
	subtitle: 'Redes Sociais'
}
describe('<FooterLinks />', () => {
	it('should render nav component', () => {
		render(<FooterLinks {...mockProps} />)
		expect(screen.getByRole('navigation')).toBeInTheDocument()
	})
	it('should render the title component correctly', () => {
		render(<FooterLinks {...mockProps} />)
		expect(
			screen.getByRole('heading', { name: /Redes Sociais/i })
		).toBeInTheDocument()
	})
	it('should render List and links component correctly', () => {
		render(<FooterLinks {...mockProps} />)
		const list = screen.getByRole('list')
		const li = screen.getByRole('listitem', { name: /item 1/i })
		expect(list).toBeInTheDocument()
		expect(li).toBeInTheDocument()
		expect(list.firstChild).toHaveAttribute('href')
		expect(li.firstChild).toHaveAttribute('rel', 'noreferrer noopener')
	})
	it('should match to snapshot', () => {
		const { container } = render(<FooterLinks {...mockProps} />)
		expect(container.firstChild).toMatchSnapshot()
	})
})
