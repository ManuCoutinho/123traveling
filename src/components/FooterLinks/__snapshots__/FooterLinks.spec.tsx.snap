// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FooterLinks /> should match to snapshot 1`] = `
<nav>
  <h2
    class="chakra-heading css-1chae5f"
  >
    Redes Sociais
  </h2>
  <ul
    class="css-119xfik"
    role="list"
  >
    <a
      href="/#"
      rel="noreferrer noopener"
    >
      <li
        aria-label="description1"
        class="css-8xctvb"
        role="listitem"
      >
        description1
      </li>
    </a>
    <a
      href="/#"
      rel="noreferrer noopener"
    >
      <li
        aria-label="description2"
        class="css-8xctvb"
        role="listitem"
      >
        description2
      </li>
    </a>
  </ul>
</nav>
`;
